hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java

```diff
@@ -0,0 +1,303 @@
+
+	public ReactiveBasicCollectionPersister(
+			Collection collectionBinding,
+			CollectionDataAccess cacheAccessStrategy,
+			PersisterCreationContext creationContext) throws MappingException, CacheException {
+		super( collectionBinding, cacheAccessStrategy, creationContext );
+		this.parameters = Parameters.instance( getFactory().getJdbcServices().getDialect() );
+	}
+
+	private ReactiveConnection getReactiveConnection(SharedSessionContractImplementor session) {
+		return ( (ReactiveConnectionSupplier) session ).getReactiveConnection();
+	}
+
+	@Override
+	protected String getSQLInsertRowString() {
+		String sql = super.getSQLInsertRowString();
+		return parameters.process( sql );
+	}
+
+	@Override
+	protected String getSQLDeleteRowString() {
+		String sql = super.getSQLDeleteRowString();
+		return parameters.process( sql );
+	}
+
+	@Override
+	protected String getSQLDeleteString() {
+		String sql = super.getSQLDeleteString();
+		return parameters.process( sql );
+	}
+
+	public CompletionStage<Void> recreateReactive(
+			PersistentCollection collection,
+			Serializable id,
+			SharedSessionContractImplementor session)
+			throws HibernateException {
+		if ( isInverse ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( !isRowInsertEnabled() ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( LOG.isDebugEnabled() ) {
+			LOG.debugf(
+					"Inserting collection: %s",
+					MessageHelper.collectionInfoString( this, collection, id, session )
+			);
+		}
+
+		CompletionStage<Integer> loop = CompletionStages.zeroFuture();
+		ReactiveConnection reactiveConnection = getReactiveConnection( session );
+
+		// create all the new entries
+		Iterator entries = collection.entries( this );
+
+		if ( entries.hasNext() ) {
+			collection.preInsert( this );
+
+			int i = 0;
+
+			while ( entries.hasNext() ) {
+				final Object entry = entries.next();
+				if ( collection.entryExists( entry, i ) ) {
+					int offset = 1;
+					Object[] paramValues = PreparedStatementAdaptor.bind(
+							st -> {
+								int loc = writeKey( st, id, offset, session );
+								if ( hasIdentifier ) {
+									loc = writeIdentifier( st, collection.getIdentifier( entry, i ), loc, session );
+								}
+								if ( hasIndex /* && !indexIsFormula */) {
+									loc = writeIndex( st, collection.getIndex( entry, i, this ), loc, session );
+								}
+								loc = writeElement( st, collection.getElement( entry ), loc, session );
+
+								collection.afterRowInsert( this, entry, i );
+							}
+					);
+					loop = loop.thenCompose( total -> reactiveConnection
+							.update( getSQLInsertRowString(), paramValues )
+							.thenApply( insertCount -> total + insertCount ) );
+				}
+			}
+		}
+
+		return loop.thenAccept( total -> LOG.debugf( "Done inserting collection: %s rows inserted", total ) );
+	}
+
+	public CompletionStage<Integer> removeReactive(Serializable id, SharedSessionContractImplementor session)
+			throws HibernateException {
```

Comment:
Shouldn't we be consistent with ORM and return `CompletionStage<Void>` instead?
See [AbstractCollectionPersister#remove](https://github.com/hibernate/hibernate-orm/blob/20f6d4f5dbccd352d5a3e219121136cc53506f89/hibernate-core/src/main/java/org/hibernate/persister/collection/AbstractCollectionPersister.java#L1206)